### ISSUE TITLE
Initial admin / setup tasks

### ISSUE THREAD

**Issue by emiliom** [BiG-CZ/notebook_data_demo#4]
@lsetiawan, please work with @ocefpaf to make decisions or assist in initial "low level" setups (TravisCI, AppVeyor, etc) for this new repo / blog system. Starting today if possible, before our long weekend starts -- assuming you're not working on Monday? Below, I'm pasting relevant text from @ocefpaf, from email exchanges, from newest to oldest. Some may already be taken care of -- @ocefpaf, maybe you can cross out those items in this comment, or just follow up highlighting what we (mainly @lsetiawan :wink:) can help you with?

> The adaptation is virtually done, so there is nothing to furiously work on :-) 

> (Although there are many details, like those listed in #3, that requires someone with more intimated knowledge on BiG-CZ.)

@ocefpaf: I'll get to that later, hopefully starting late next week. Again, there's no rush, we're not publicizing or linking to it from anywhere, any time soon!

> I guess I was waiting for your input on a strategy to enable the CIs (see ou previous e-mails). I will go ahead and enable Travis-CI myself since that one has the team-like permissions from GitHub, that way I can continue working on the auto-publishing. However, we need a better strategy regarding the multiple AppVeyor services.

------------------------------------

> Each CI service and deployment keys, via TOKENS, works in a different way. So there is no "ODM2" account for Travis-CI and AppVeyor. There are users accounts with rights to access certain GitHub content via permissions.
> 
> For example, to activate Travis-CI and AppVeyor one must be the repository owner, not necessarily the org owner. Travis-CI will propagate that to all owners and everybody will be able to access the Travis-CI config as long as they have access to the repo. AppVeyor is slightly different though, it is attached to the account that enabled it instead and each user must be made "owner" of the AppVeyor instance separately.
>
> Anyways, AppVeyor is not essential at the moment. I like to enable it to catch Windows problems soon.
>
> You can, after creating the repository, enable Travis-CI for it. After that we will still need to create the deployment key, but I can guide you through that later when the time comes. (With the repo and Travis-CI enable I will have plenty to do already.)

------------------------------------

> Now I need to start configuring the auto publishing and testing, but that tied to Travis-CI and AppVeyor accounts with tokens set to the org/repo.

**Comment by lsetiawan** [comment]
Woo hoo! Another project with @ocefpaf 😄. I'm here for whatever you need. In the mean time, I'll try to wrap my head around what you're doing. @emiliom has explained to me a bit. Here goes! 

FYI for @emiliom I am working on Monday since it's not a holiday 😛

**Comment by emiliom** [comment]
@ocefpaf, is there anything here that isn't taken care of yet? Or can we close this issue?

**Comment by ocefpaf** [comment]
We don't have an AppVeyor account yet. That is not urgent and maybe we don't really need it.
I usually like to enable AppVeyor testing to be sure that Windows user are covered.

**Comment by emiliom** [comment]
> I usually like to enable AppVeyor testing to be sure that Windows user are covered.

In this context, what does this mean? That the notebooks themselves are tested to make sure they run on Windows?

**Comment by ocefpaf** [comment]
> That the notebooks themselves are tested to make sure they run on Windows?

Exactly. It provides a real case integration test for the notebook code, the software installation, and the services used. We usually uncover tons of hidden bugs when doing that, from simple hard-coded paths, to non-portable code syntax.

**Comment by emiliom** [comment]
>> That the notebooks themselves are tested to make sure they run on Windows?
>
> Exactly. It provides a real case integration test for the notebook code, the software installation, and the services used.

Ah. Great.

**Comment by emiliom** [comment]
I guess I should've separated out the tasks from #3 ... The second one belongs here:

Is the CI working (both Travis and AppVeyor)? Looks like they're not, based on the status icons (I forget the name of those things!) on the README.md

**Comment by ocefpaf** [comment]
Checking the CIs today. Not sure what is going on with Travis-CI (AppVeyor is just a placeholder and will be replaced by the "ODM2 account" soon.)

**Comment by emiliom** [comment]
Thanks!

**Comment by ocefpaf** [comment]
@emiliom I added the `odm2bot` user to the `ODM2` organization and attached all AppVeyor jobs to it.
I cannot do the same for the `BiG-CZ` org b/c I do not have adim rights here.

Can you do that? I believe the `odm2bot` does not require full rights to the org. I am unclear to what levels GitHub offers these days but a "collaborator" should suffice.

**Comment by emiliom** [comment]
@ocefpaf I've changed your privileges on the BiG-CZ org to full admin rights.

**Comment by ocefpaf** [comment]
> @ocefpaf I've changed your privileges on the BiG-CZ org to full admin rights.

Thanks. I'm activating the bot account now.

**Comment by emiliom** [comment]
great
